### ISSUE TITLE
Increase accessibility for users with light/flashing sensitivity

### ISSUE THREAD

**Issue by TimMonko** [napari/napari#7433]
## 🧰 Task

I was unable to find any other issues related to light-sensitivity and accessibility, so I thought I would make this issue based on my own experiences as someone with light sensitivity. This issue is also motivated by a brief discussion with @psobolewskiPhD and @jni in #7427, relating to altering the default for `viewer.screenshot()`-type functions to reduce risk of issue for light sensitive users. In that PR, it was proposed to change the default to drop the flashing of the viewer, which could be important if a user naively triggers repetitive screenshots via script, resulting in high frequency flashing. 

Here is a list of tasks, based on some things I've observed as possible issues. Would love for people to help work on this or offer any other things to be changed. 

- [ ] Add a `light sensitivity adjustments` option to Preferences. This setting could trigger a series of changes to napari's GUI:
    - [ ] Turn off the grayscale flashing of `viewer.screenshot()`-like GUI commands, by setting GUI-triggered commands to `flash=False`
    - [ ] Limit the speed of playback, perhaps to something like 30fps. This would limit rapid blinking if there is high Z or T variability. Currently, the default playback speed should be ok, but this would hard cap the upper bound. 
    - [ ] Fade the transition to new contrast limits. Instead of an immediate switch, fade to the new limits or put a constraint on the %change of the contrast adjustment. This is again important when going through multiple Z or T and contrast is set to continuous. 
    - [ ] Fade in layers added to the viewer. Sometimes, many layers get dumped in to the viewer asynchronously, or as-loaded. One option would be to fade in each layer to 'blend' new layers rather than cause a rapid change with a transition to the whole layer (especially if it uses translucent blending or something). Another option would be to literally sleep the addition of each layer, but I think that's a terrible idea.
- [ ] Change the default behavior of `viewer.screenshot()`-like funs to have `flash=False` as discussed in #7427. Thus, when triggered from scripts it reduces the risk of inadvertently flashing the users. GUI-triggered behavior would have `flash=True` to maintain the current feedback for users. (And this would be overwritten if a user opts in to the above `light sensitivity adjustments`)

**Comment by Czaki** [comment]
@TimMonko I like this proposal. However, I think that implementing it will require input from a person with light-sensitivity. 

Did you plan to make, a PR which will address (at least part of these points) this? Maybe we should provide some feedback on how it should be implemented? 

Or you (or someone who you know) could test some solution if someone decides to implement it?

**Comment by TimMonko** [comment]
@Czaki, thank you for your feedback! 
My goal is to become a regular contributor to napari, so I am happy to work on implementing these. However, it's quite slow for me since I am self taught, and working through how napari internally connects with itself is still a challenge for me. 

I am planning to specifically get a PR in for the `screenshot(flash=False)` default change soon, but some of the other changes would definitely benefit from guidance. How do people usually get help starting to contribute further? Zulip? Community meetings? 

And of course, I'm happy to be the tester. I mean, as long as we do it right, it won't trigger my light-sensitivity lol (I finally remember the word is photosensitive...) Cheers-

**Comment by Czaki** [comment]
> I am planning to specifically get a PR in for the `screenshot(flash=False)` default change soon, but some of the other changes would definitely benefit from guidance. 

I suggest implementing this by adding a proper entry in settings (I suggest apperance https://github.com/napari/napari/blob/main/napari/settings/_appearance.py) 

And then change the default value to `None`
Then use something like:
``` 
def screenshot(flash=None):
    if flash is None: 
        flash = not get_settings().apperance.light_sensitivity
    ...
```

> How do people usually get help starting to contribute further? Zulip? Community meetings?

Depend on the person. Sometimes just opening PR. Sometimes start asking questions on Zulip. Sometimes Join community meeting and ask questions there. It is sometimes possible to have a call with one or more core-dev to have time to explain all doubts 

For fading changes, there is a need to think how to do it. As using a timer per change will introduce multiple problems. We should create some event queue that will be triggered per tick that we need to adjust.
